rosidl_generator_dotnet/cmake/rosidl_generator_dotnet_generate_interfaces.cmake

```diff
@@ -215,9 +225,13 @@ if(_generated_c_ts_files)
     ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
     ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_dotnet
   )
-
-  rosidl_target_interfaces(${_target_name}
-    ${rosidl_generate_interfaces_TARGET} rosidl_typesupport_c)
+  
+  if(ros2_distro IN_LIST humble_distro_and_newer)
+    rosidl_get_typesupport_target(c_typesupport_target "${rosidl_generate_interfaces_TARGET}" "rosidl_typesupport_c")
+    target_link_libraries(${_target_name} "${c_typesupport_target}")
+  else()
+    rosidl_target_interfaces(${_target_name} ${rosidl_generate_interfaces_TARGET} rosidl_typesupport_c)
+  endif()
```

Comment:
Not sure if I got this right, could you explain why this change is necessary?
Seems like the way interfaces are generated changed in humble?

Reply:
Yeah is indeed the deprecation of that function